crates/ty_python_semantic/src/types/class.rs

```diff
@@ -1097,11 +1097,11 @@ impl MethodDecorator {
     }
 }
 
-/// Metadata regarding a dataclass field/attribute.
+/// Metadata regarding a dataclass field/attribute or a `TypedDict` "item" / key-value pair.
 #[derive(Debug, Clone, PartialEq, Eq)]
-pub(crate) struct DataclassField<'db> {
+pub(crate) struct Field<'db> {
     /// The declared type of the field
-    pub(crate) field_ty: Type<'db>,
+    pub(crate) declared_ty: Type<'db>,
```

Comment:
clippy forced me to rename this attribute, as it would otherwise start with the struct name..